src/zenml/pipelines/base_pipeline.py

```diff
@@ -260,7 +271,40 @@ def requirements(self) -> Set[str]:
                     f"'{integration_name}'."
                 ) from e
 
-        if self.requirements_file and fileio.exists(self.requirements_file):
+        if (
+            isinstance(self._requirements, str)
+            and self._requirements
+            and fileio.exists(self._requirements)
+        ):
+            with fileio.open(self._requirements, "r") as f:
+                requirements.update(
+                    {
+                        requirement.strip()
+                        for requirement in f.read().split("\n")
+                    }
+                )
+        elif isinstance(self._requirements, str) and self._requirements:
+            root = str(Repository().root)
```

Comment:
If we keep this logic of falling back to this default file (which I'm not a huge fan of), I'd suggest we use 
`zenml.utils.source_utils.get_source_root_path()` instead of the repo root. This will return either 
- the repo root if a `.zen` folder exists in one of the parent directories
- the parent directory of the python file you're running

Reply:
Following a discussion with @htahir1, I've removed this whole condition and made a ticket for someone to add it back in with the full logic of what needs to happen.